logger/src/lib.rs

```diff
@@ -664,7 +664,7 @@ impl Logger {
             }
         } else {
             METRICS.logger.metrics_fails.inc();
-            return Err(LoggerError::LogMetricFailure(
+            return Err(LoggerError::NeverInitialized(
```

Comment:
This change doesn't seem to belong to the same commit as the other ones. I have a nit here: previously in the code, `NeverIntialized` is an error used when initialization of the logger fails. From what I understand, here we want to throw an error when logging fails because the logger was not initialized. I suggest adding another error type for this scenario called `Uninitialized` and maybe rename the `NeverInitialized` to `IntiailizationFailed`. This change would make the code a bit easier to read.